### PR TITLE
#87 - add pagination

### PR DESCRIPTION
**Pull request by MrHDOLEK**
Should close https://github.com/blumilksoftware/meetup/issues/87

### CODE REVIEW

File: resources/views/components/input-limits-pagination.blade.php [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<div class="flex gap-5">
+    <select class="block appearance-none w-full bg-white border border-gray-400 hover:border-gray-500 px-4 py-2 pr-8 rounded shadow leading-tight focus:outline-none focus:shadow-outline" onchange="window.location.href=this.options[this.selectedIndex].value;">>
+        <option value="{{ route((string)request()->route()->getName(),['limit'=> 10 ]) }}">Limit: 10</option>
+        <option value="{{ route((string)request()->route()->getName(),['limit'=> 30 ]) }}">Limit: 30</option>
+        <option value="{{ route((string)request()->route()->getName(),['limit'=> 50 ]) }}">Limit: 50</option>
+        <option value="{{ route((string)request()->route()->getName(),['limit'=> 100 ]) }}">Limit: 100</option>
+    </select>
+</div>
```
Comment (kamilpiech97) [inline resources/views/components/input-limits-pagination.blade.php:8]:
```suggestion
</div>

```

File: resources/views/meetups/index.blade.php [hunk 2]
```diff
@@ -7,7 +7,7 @@
             @auth
                 <a href="{{ route('meetups.create') }}">Create new meetup</a>
             @endauth
-            @if ($meetups->count())
+        @if ($meetups->count())
                 @foreach ($meetups as $meetup)
```
Comment (kamilpiech97) [inline resources/views/meetups/index.blade.php:11]: Use `forelse`.
```suggestion
                @forelse ($meetups as $meetup)
```

File: src/Http/Controllers/MeetupController.php [hunk 3]
```diff
@@ -20,9 +21,10 @@ public function __construct(
         protected StoreFileService $storeFileService,
     ) {}
 
-    public function index(): View
+    public function index(PaginationRequest $request): View
     {
-        $meetups = Meetup::query()->latest()->with(["user"])->paginate(20);
+        $paginationLimit = $request->input("limit");
```
Comment (kamilpiech97) [inline src/Http/Controllers/MeetupController.php:26]: What about starting value?

**Review by kamilpiech97 — CHANGES_REQUESTED** [review]

File: src/Http/Controllers/MeetupController.php (hunk 3 above)
Comment (MrHDOLEK) [reply src/Http/Controllers/MeetupController.php:26]: if limit is empty to pagination is 15

**Review by kamilpiech97 — APPROVED** [review]

File: resources/views/components/input-limits-pagination.blade.php [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<div class="flex gap-5">
+    <select class="block appearance-none w-full bg-white border border-gray-400 hover:border-gray-500 px-4 py-2 pr-8 rounded shadow leading-tight focus:outline-none focus:shadow-outline" onchange="window.location.href=this.options[this.selectedIndex].value;">>
+        <option value="{{ route((string)request()->route()->getName(),['limit'=> 10 ]) }}">Limit: 10</option>
+        <option value="{{ route((string)request()->route()->getName(),['limit'=> 30 ]) }}">Limit: 30</option>
+        <option value="{{ route((string)request()->route()->getName(),['limit'=> 50 ]) }}">Limit: 50</option>
+        <option value="{{ route((string)request()->route()->getName(),['limit'=> 100 ]) }}">Limit: 100</option>
```
Comment (krzysztofrewak) [inline resources/views/components/input-limits-pagination.blade.php:6]:
```suggestion
        <option value="{{ route((string)request()->route()->getName(), ['limit'=> 10 ]) }}">Limit: 10</option>
        <option value="{{ route((string)request()->route()->getName(), ['limit'=> 30 ]) }}">Limit: 30</option>
        <option value="{{ route((string)request()->route()->getName(), ['limit'=> 50 ]) }}">Limit: 50</option>
        <option value="{{ route((string)request()->route()->getName(), ['limit'=> 100 ]) }}">Limit: 100</option>
```

Comment (krzysztofrewak) [reply resources/views/components/input-limits-pagination.blade.php:6]: And do you really need to call router four times? Maybe you could move this value to controller or view composer? Or maybe `#` would be enough?

File: src/Http/Requests/PaginationRequest.php [hunk 5]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Blumilk\Meetup\Core\Http\Requests;
+
+use Illuminate\Foundation\Http\FormRequest;
+use Illuminate\Validation\Rule;
+
+class PaginationRequest extends FormRequest
```
Comment (krzysztofrewak) [inline src/Http/Requests/PaginationRequest.php:10]: Are you sure that's a good idea? What if you would like to add pagination to some controller's method what would require other form request? And do you really think that invalid pagination limit should crash application?

**Review by krzysztofrewak — CHANGES_REQUESTED** [review]

File: database/seeders/DummyDataSeeder.php [hunk 6]
```diff
@@ -20,7 +19,7 @@ public function run(): void
         $user = User::factory([
             "name" => "Admin",
             "email" => "admin@example.com",
-            "password" => Hash::make("password"),
+            "password" => "password",
```
Comment (Majkulnn) [inline database/seeders/DummyDataSeeder.php:22]: write back Hash:make here

**Review by kamilpiech97 — DISMISSED** [review]

**Review by Majkulnn — APPROVED** [review]

**Review by krzysztofrewak — APPROVED** [review]

**Pull request merged**